pkg/client/restclient/request.go

```diff
@@ -811,7 +813,14 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 			r.backoffMgr.UpdateBackoff(r.URL(), err, resp.StatusCode)
 		}
 		if err != nil {
-			return err
+			operr, ok := err.(*net.OpError)
+			if ok && operr.Err.Error() == syscall.ECONNRESET.Error() {
```

Comment:
Sure - I will play with that. Just wanted to get your buy in around the idea. WIll try to clean this PR up tomorrow.